dcos_test_utils/dcos_api_session.py

```diff
@@ -323,12 +323,28 @@ def _wait_for_srouter_slaves_endpoints(self):
                     retry_on_result=lambda r: r is False,
                     retry_on_exception=lambda _: False)
     def _wait_for_metronome(self):
+        # Although this is named `wait_for_metronome`, some of the waiting
+        # done in this function is, implicitly, for Admin Router.
         r = self.get('/service/metronome/v1/jobs')
-        # 500 and 504 are the expected behavior of a service
-        # backend that is not up and running.
-        if r.status_code == 500 or r.status_code == 504:
-            log.info("Metronome gateway timeout, continue waiting for backend...")
+        expected_error_codes = {
+            404: ('It may be the case that Admin Router is returning a 404 '
+                  'despite the Metronome service existing because it uses a cache. '
+                  'This cache is updated periodically.'),
+            503: ('The cache is not working. This means that either Marathon '
+                  'or Mesos is not yet available.'),
+            504: ('Metronome is returning a Gateway Timeout Error.'
+                  'It may be that the service is still starting up.')
+        }
+
+        if r.status_code in expected_error_codes or r.status_code > 500:
+            error_message = expected_error_codes.get(r.status_code)
+            if error_message:
+                log.info(error_message)
+            log.info('Continuing to wait for Metronome')
+            log.info('Response body:')
+            log.info(r.content.decode())
```

Comment:
I saw that used elsewhere in this repository, but so is `r.text` so I have changed it.